storage/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerDbManager.java

```diff
@@ -280,6 +298,169 @@ public void assignSviToDevice(DataSource ds, UUID guid, String sviString) {
+    for (int i = 1; i < modSequence.size(); i++) {
+      if (!(modSequence.get(i - 1).equals(modSequence.get(i)))) {
+        sb.append(modSequence.get(i));
+        sb.append(":active=activate_mod,");
+      }
+      sb.append(sviEntries.get(i) + ",");
+    }
+
+    sb.deleteCharAt(sb.length() - 1);
+    return sb.toString();
+  }
+
+  /**
+   * Determines if wget module is included in svi.
+   *
+   * @param sviString svi mapping string
+   * @return checks if wget module entries are present in the service info string
+   */
+  public boolean checkWgetInSviString(String sviString) {
+
+    List<String> sviEntries = Arrays.asList(sviString.split(","));
+
+    for (String sviEntry : sviEntries) {
+      String[] sviEntryArray = sviEntry.split(SVI_ENTRY_DELIMETER);
+      String[] modMsgDelimeted = sviEntryArray[0].split(SVI_MODMSG_DELIMETER);
+      if (modMsgDelimeted[0].equals("sdo_wget")) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if content verification is enabled for wget file content.
+   *
+   * @param ds Datasource
+   * @return checks wget verification enable status
+   */
+  public boolean isWgetVerificationEnabled(DataSource ds) {
+    boolean isWgetContentVerificationEnabled = false;
+
+    String sql = "SELECT WGET_SVI_MOD_VERIFICATION FROM TO2_SETTINGS WHERE ID = 1";
+    try (Connection conn = ds.getConnection();
+        PreparedStatement pstmt = conn.prepareStatement(sql)) {
+      try (ResultSet rs = pstmt.executeQuery()) {
+        if (rs.next()) {
+          String colValue = rs.getString(1);
+          if (colValue.equalsIgnoreCase("true")) {
+            isWgetContentVerificationEnabled = true;
+          } else if (colValue.equalsIgnoreCase("false")) {
+            isWgetContentVerificationEnabled = false;
+          } else {
+            throw new RuntimeException(
+                "Invalid entry in TO2_SETTINGS for WGET_CONTENT_VERIFICATION field");
+          }
+        }
+      }
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+
+    return isWgetContentVerificationEnabled;
+  }
+
+  /**
+   * Calculated sha384 hash of file content to be transferred using wget:url. Adds the hash entry to
+   * OWNER_SERVICEINFO table Includes appropriate mapping in the sviString for corresponding hash
+   * file.
+   *
+   * @param ds datasource
+   * @param sviString svi mapping string to be updated
+   * @return updated sviString with activate module entries
+   */
+  public String insertWgetContentHash(DataSource ds, String sviString) {
+    StringBuilder sb = new StringBuilder();
+    List<String> sviEntries = Arrays.asList(sviString.split(","));
+    List<String> newSviEntries = new ArrayList<>();
+
+    for (String sviEntry : sviEntries) {
+      String[] sviEntryArray = sviEntry.split(SVI_ENTRY_DELIMETER);
+
+      // Get the filename whose hash needs to be calculated.
+      if (sviEntryArray[0].trim().equals("sdo_wget:filename")) {
```

Comment:
consider using a constant for this literal

Reply:
Done